code/game/machinery/cryopod.dm

```diff
@@ -0,0 +1,430 @@
+	onclose(user, "cryopod_console")
+
+/obj/machinery/computer/cryopod/Topic(href, href_list)
+	if(..())
+		return 1
+
+	var/mob/user = usr
+
+	add_fingerprint(user)
+
+	if(href_list["log"])
+
+		var/dat = "<b>Recently stored [storage_type]</b><br/><hr/><br/>"
+		for(var/person in frozen_crew)
+			dat += "[person]<br/>"
+		dat += "<hr/>"
+
+		user << browse(dat, "window=cryolog")
+
+	if(href_list["view"])
+		if(!allow_items) return
+
+		var/dat = "<b>Recently stored objects</b><br/><hr/><br/>"
+		for(var/obj/item/I in frozen_items)
+			dat += "[I.name]<br/>"
+		dat += "<hr/>"
+
+		user << browse(dat, "window=cryoitems")
+
+	else if(href_list["item"])
+		if(!allowed(user))
+			to_chat(user, "<span class='warning'>Access Denied.</span>")
+			return
+		if(!allow_items) return
+
+		if(frozen_items.len == 0)
+			to_chat(user, "<span class='notice'>There is nothing to recover from storage.</span>")
+			return
+
+		var/obj/item/I = input(user, "Please choose which object to retrieve.","Object recovery",null) as null|anything in frozen_items
+		if(!I)
+			return
+
+		if(!(I in frozen_items))
+			to_chat(user, "<span class='notice'>\The [I] is no longer in storage.</span>")
+			return
+
+		visible_message("<span class='notice'>The console beeps happily as it disgorges \the [I].</span>")
+
+		I.forceMove(get_turf(src))
+		frozen_items -= I
+
+	else if(href_list["allitems"])
+		if(!allowed(user))
+			to_chat(user, "<span class='warning'>Access Denied.</span>")
+			return
+		if(!allow_items) return
+
+		if(frozen_items.len == 0)
+			to_chat(user, "<span class='notice'>There is nothing to recover from storage.</span>")
+			return
+
+		visible_message("<span class='notice'>The console beeps happily as it disgorges the desired objects.</span>")
+
+		for(var/obj/item/I in frozen_items)
+			I.forceMove(get_turf(src))
+			frozen_items -= I
+
+	updateUsrDialog()
+	return
+/* Should more cryos be buildable?
+    /obj/item/circuitboard/cryopodcontrol
+	name = "Circuit board (Cryogenic Oversight Console)"
+	build_path = "/obj/machinery/computer/cryopod"
+	origin_tech = "programming=1"
+*/
+//Cryopods themselves.
+/obj/machinery/cryopod
+	name = "cryogenic freezer"
+	desc = "Suited for Cyborgs and Humanoids, the pod is a safe place for personnel affected by the Space Sleep Disorder to get some rest."
+	icon = 'icons/obj/machines/cryopod.dmi'
+	icon_state = "cryopod-open"
+	density = TRUE
+	anchored = TRUE
+	state_open = TRUE
+
+	var/on_store_message = "has entered long-term storage."
+	var/on_store_name = "Cryogenic Oversight"
+
+	// 3 minutes-ish safe period before being despawned.
+	var/time_till_despawn = 3 * 600 // This is reduced to 30 seconds if a player manually enters cryo
+	var/despawn_world_time = null          // Used to keep track of the safe period.
+
+	var/obj/machinery/computer/cryopod/control_computer
+	var/last_no_computer_message = 0
+
+	// These items are preserved when the process() despawn proc occurs.
+	var/static/list/preserve_items = list(
```

Comment:
where is this list coming from, objective items? create it dynamically

Reply:
could you be a little more specific?

Reply:
Not all of these, but a lot, seem to be from theft objectives, which are all written in code and can be retrieved at runtime to create this list once in Initialize (or as `GLOBAL_LIST_INIT(cryopod_preserve_items, create_cryopod_preserve_items())`). On second glance, however, I really don't like the whack-a-mole game this creates where every important item will have to add to this list. I think I'd much rather the players have to drop off the items themselves, and be completely naked in the pods (while remembering things like storage implants exist).

Reply:
i'd rather write in even more preserved objects than have players strip tbh

Reply:
This isn't a maintainable strategy.

Reply:
I see your point. I'm thinking doing this by inverting this list: Saving ALL items on the mob, but adding clothing to the do not preserve list. This way the machine would be stripping them

Reply:
attempted to invert this list but ended up filling the ```do_not_preserve``` list just as much, ex: pdas, ids, clothing, etc. The list ended up getting so large there was basically no point (nor dynamism) to it. So I'm out of ideas and not talented enough to mess with objective.dm

Reply:
Like I said, I prefer you don't have this list at all--it's just more crud we'll have to maintain. Drop all their items when they go in, or require they be naked..

Reply:
in this case you either end with piles on piles of items or players outright not doing it because they have to do x-action on top of cryoing